internal/worker/task.go

```diff
@@ -82,6 +83,35 @@ func (worker *Worker) startTask(
 	worker.logger.Infof("started task %d", agentAwareTask.TaskId)
 }
 
+func (worker *Worker) getInstance(ctx context.Context, isolation *api.Isolation) (abstract.Instance, error) {
```

Comment:
Let's add some OTEL metric on hit rate.

Reply:
Added in https://github.com/cirruslabs/cirrus-cli/pull/700/commits/62c104a0b90737ad0ce238f0b3e3238cc77c2c26.